network/network-service/client/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use aptos_config::network_id::PeerNetworkId;
+use aptos_types::PeerId;
+use async_trait::async_trait;
+use network::{
+    application::{
+        interface::{MultiNetworkSender, NetworkInterface},
+        storage::{LockingHashMap, PeerMetadataStorage},
+    },
+    error::NetworkError,
+    peer_manager::{ConnectionRequestSender, PeerManagerRequestSender},
+    protocols::network::{
+        AppConfig, ApplicationNetworkSender, NetworkSender, NewNetworkSender, RpcError,
+    },
+    ProtocolId,
+};
+use network_service_types::{
+    NetworkServiceError, NetworkServiceMessage, NetworkServiceRequest, NetworkServiceResponse,
+};
+use std::{sync::Arc, time::Duration};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Aptos network rpc error: {0}")]
+    RpcError(#[from] RpcError),
+
+    #[error("Error from remote network service: {0}")]
+    NetworkServiceError(#[from] NetworkServiceError),
+}
+
+/// The interface for sending network service requests and querying
+/// network peer information.
```

Comment:
Sadly this file is all boilerplate that needs to be copied between each service at this point.

Reply:
One day we'll get there! 🚀